### PR TITLE
DataLake Knack conversion

### PR DESCRIPTION
**Pull request by tjprescott**
Implements #4931 and #4932.

Tests not included as part of CI as there are some issues with some of them (possibly service flakiness).

### CODE REVIEW

File: src/azure-cli-testsdk/azure/cli/testsdk/base.py [hunk 1]
```diff
@@ -70,32 +70,49 @@ def __init__(self, method_name, config_file=None, recording_name=None,
...
-                patch_time_sleep_api,
-                patch_long_run_operation_delay,
-                patch_load_cached_subscriptions,
-                patch_retrieve_token_for_user,
-                patch_progress_controller,
-            ],
+            recording_processors=_merge_lists(default_recording_processors, recording_processors),
+            replay_processors=_merge_lists(default_replay_processors, replay_processors),
+            recording_patches=_merge_lists(default_recording_patches, recording_patches),
+            replay_patches=_merge_lists(default_replay_patches, replay_patches),
```
Comment (derekbekoe) [inline src/azure-cli-testsdk/azure/cli/testsdk/base.py]: Prev. is was `or`. Now you are merging.
Is this intended and do we want to support merging?

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/commands.py [hunk 2]
```diff
@@ -172,7 +172,7 @@ def load_command_table(self, _):
     # Resource feature commands
     if self.supported_api_version(min_api='2017-05-10'):
         feature_table_transform = '{Name:name, RegistrationState:properties.state}'
-        with self.command_group('feature', resource_feature_sdk) as g:
+        with self.command_group('feature', resource_feature_sdk, client_factory=cf_features) as g:
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/commands.py]: Why this change?
If it is needed, I guess you can remove `client_factory=cf_features` below?

File: src/azure-cli-testsdk/azure/cli/testsdk/base.py (hunk 1 above)
Comment (tjprescott) [reply src/azure-cli-testsdk/azure/cli/testsdk/base.py]: Yes, this was very intended. The only reason is was "or" for so long was because no one ever tried to use it... we just always added any processors we needed to the core testbase (which isn't really a good idea...)

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/commands.py (hunk 2 above)
Comment (tjprescott) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/commands.py]: Yeah, pretty much so I could remove the kwarg in the lines below.
